src/Elements/junction.cpp

```diff
@@ -82,6 +83,7 @@ void Junction::initialize(Network* nw)
 {
     head = elev + (pFull - pMin) / 2.0;;
     quality = initQual;
+    if (qualSource) qualSource->quality = quality;
```

Comment:
initialize the source quality

Reply:
There is no need to initialize `qualSource->quality`. This variable is assigned locally in `QualSource::getQuality`.